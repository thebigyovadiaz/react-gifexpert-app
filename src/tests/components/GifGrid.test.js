import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");

describe("Pruebas Unitarias de <GifGrid />", () => {
  const category = "Goku";

  test("Debe crear el snapshot del componente", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar los items al cargar imÃ¡genes useFetchGifs", () => {
    const gifs = [
      {
        id: "ABC",
        url: "https://localhost/hola.jpg",
        title: "Hola",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    // expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
