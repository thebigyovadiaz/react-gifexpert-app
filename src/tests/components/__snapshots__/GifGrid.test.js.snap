// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas Unitarias de <GifGrid /> Debe crear el snapshot del componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInLeftBig"
  >
    Goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
